### DISCUSSION TITLE
tlmWrite in Drv::LinuxSerialDriver

### DISCUSSION THREAD

**Discussion by danjwait** [nasa/fprime#1404] [Q&A] [answered]
Trying to debug a serial connection I noticed the serial component telemetry for DR_BytesRecv and DR_BytesSent called out in [Telemetry.fppi](https://github.com/nasa/fprime/blob/devel/Drv/LinuxSerialDriver/Telemetry.fppi) weren't updating on the GDS, though I think per the .fpp files for the LinuxSerialDriver and the Pattern graph specifiers in the topology.fpp file I have (and after running fpp-check) those ports are setup and connected, there's just no data moving across.

Looking at [LinuxSerialDriverComponentImpl.cpp](https://github.com/nasa/fprime/blob/devel/Drv/LinuxSerialDriver/LinuxSerialDriverComponentImpl.cpp) and [LinuxSerialDriverComponentImplCommon.cpp](https://github.com/nasa/fprime/blob/devel/Drv/LinuxSerialDriver/LinuxSerialDriverComponentImplCommon.cpp) I don't see something like a `this->tlmWrite_DR_BytesRecv(<bytesRx>)` or `this->tlmWrite_DR_BytesRecv(<bytesTx>)` 

I think if I want to see that telemetry, I'd need to add those tlmWrites maybe in LinuxSerialDriverComponentImpl.cpp , maybe:
 - `this->tlmWrite_DR_BytesSent(xferSize)`  in serialSend_handler() 
 - `this->tlmWrite_DR_BytesRecv (totalSizeRead )`  in serialReadTaskEntry()
 
I think those values will end up being something like per read/writes as opposed to cumulative, but I think that may still help figure out if the instance is working. If that's the right thing to do, should I try it out and make a pull request?

Thanks!

**Comment by timcanham** [comment, marked as answer]
Yes, good find. We knew about it, but hadn't fixed it yet. If you write the value each time bytes are transmitted or received, that could result in a a lot of calls and cause performance issues. What I would do is add a [`Svc::Sched`](https://github.com/nasa/fprime/tree/devel/Svc/Sched) port and handler that could periodically be polled for the counts if that was desired. Then do a PR!

**Reply by danjwait** [reply]
Thanks! I'll take a look and if I get it to work I'll open the PR
